File: fastapi-starter/app/api/health.py
"""Health check endpoints."""

from datetime import datetime

from fastapi import APIRouter

from app.core.config import settings

router = APIRouter()


@router.get("/health")
async def health_check():
    """Basic health check endpoint."""
    return {
        "status": "healthy",
        "timestamp": datetime.utcnow().isoformat(),
        "version": settings.APP_VERSION,
    }


@router.get("/health/ready")
async def readiness_check():
    """Readiness probe for Kubernetes."""
    # Add database/cache connectivity checks here
    checks = {
        "app": True,
    }

    if settings.DATABASE_URL:
        # TODO: Add actual database check
        checks["database"] = True

    if settings.REDIS_URL:
        # TODO: Add actual Redis check
        checks["redis"] = True

    all_healthy = all(checks.values())

    return {
        "status": "ready" if all_healthy else "not_ready",
        "checks": checks,
        "timestamp": datetime.utcnow().isoformat(),
    }


@router.get("/health/live")
async def liveness_check():
    """Liveness probe for Kubernetes."""
    return {"status": "alive"}

File: express-starter/src/routes/health.ts
import { Router } from 'express';
import { config } from '../config/env.js';

const router = Router();

router.get('/', (_req, res) => {
  res.json({
    status: 'healthy',
    timestamp: new Date().toISOString(),
    version: config.APP_VERSION,
  });
});

router.get('/ready', (_req, res) => {
  // Add database/cache connectivity checks here
  const checks: Record<string, boolean> = {
    app: true,
  };

  if (config.DATABASE_URL) {
    // TODO: Add actual database check
    checks.database = true;
  }

  if (config.REDIS_URL) {
    // TODO: Add actual Redis check
    checks.redis = true;
  }

  const allHealthy = Object.values(checks).every(Boolean);

  res.status(allHealthy ? 200 : 503).json({
    status: allHealthy ? 'ready' : 'not_ready',
    checks,
    timestamp: new Date().toISOString(),
  });
});

router.get('/live', (_req, res) => {
  res.json({ status: 'alive' });
});

export default router;
